stories/button.story.jsx

```diff
@@ -57,130 +56,117 @@ storiesOf("Buttons", module).add(
 
       <h2>Primary Call to Action</h2>
       <div className="list">
-        <Button onClick={() => click("button clicked!")}>Call to Action</Button>
-        <Button onClick={() => click("button clicked!")}>
-          <Icon name="call-contact" />
-          Call to Action
+        <Button>
```

Comment:
Correct me if I am wrong here,
 I have the feeling here we just write plain html and using styled components we parse css classes on the tags.
Although it gives me the feeling this is not a component. As all the logic of the component is being repeated every time we want to create a Button or a Button that is an anchor under the hood.
The `<Button >` in this case is just css classes.
 
I believe we abstract this so much that we repeat ourselves every-time we want to create a Button. 

Is it that I get what u explained wrong here?

Reply:
So that's not what I explained. I said if it were me I'd still have a general use button component for the common case and a CustomButton component that uses button styles but does not enforce its children. That's option one. Option two would be to have a Button and a AnchorButton with  specified props. For example, we would enforce for the anchor button to always have the rel property set, and only allow a few other properties we specifically design. **We should not**, however, pass in a {...props} object that allows everyone to pass in whatever they want. I know that this is convenient and seems to solve the problem of catching all use cases but it has definite downsides. I can write a blog post about the downside but it seems pretty obvious to me - we're limiting our ability to change the component for the future, we're not allowing an explicit api to be set for users of the button, we're not guaranteeing types or checking for safety, etc. Plus there could even be considerations on how do React's optimizations work with dynamic props such as this (perhaps it compiles it down correctly and then can run its optimizations).

As a general rule always try to limit the complexity of a thing you want to use. It doesn't matter if its a component, function, variable, etc. Always try to guarantee a specific state that can 100% persist. This same idea applies to not using boolean toggles. That is, the various complexities and states that can occur with toggle = !toggle are greater and more unpredictable than () => toggle = false; () => toggle = true.

Reply:
Which option do we want to go with ?

Reply:
I will go with what u suggested.

Reply:
I usually go with option 2 and if absolutely necessary option 1

Reply:
Forgot about a third option - which is a sort of pseudo OOP, where you create a base component (say CustomButton), then create two components that extend the base, Button and AnchorButton. Although not sure if its really necessary in this case, but felt like noting it was relevant.